study.md

```diff
@@ -246,5 +259,5 @@ Pick a few design heuristics. Describe some common, named refactorings you can
 use approach a good design.
 
 ```md
-<!-- your answer here -->
+I feel that I could use the guide line of methods should no longer than five lines of code. I could see how my methods could possibly break that rule. I would also adopt that passing more than four parameters or a big hash could indeed be bad/dangerous. All of the heuristic guide lines should always be followed when possible.
```

Comment:
Great start. What would you use to DRY up duplicated logic? Maybe "Extract Method". Have a look at some named refactorings, like "Extract Method", to level up.